### PR TITLE
Editorial: `name` not allowed on a `li` element

### LINKED ISSUE #214: `name` not allowed on a `li` element
Ref: https://validator.w3.org/nu/?showsource=yes&doc=https%3A%2F%2Fw3c.github.io%2Faccname%2F

The HTML checker complains that there is a `name` attribute on a `li` element. Can we use another method for this that passes the HTML validator?

### PR DESCRIPTION
**Pull request by jnurthen**
Fixes #214

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/accname/pull/219.html" title="Last updated on Nov 4, 2023, 3:34 AM UTC (ab9b86f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/accname/219/3e8f6d4...ab9b86f.html" title="Last updated on Nov 4, 2023, 3:34 AM UTC (ab9b86f)">Diff</a>

### CODE REVIEW

**Review by cookiecrook — CHANGES_REQUESTED** [review]
If you move the `id` to the `em`, you won't need the bogus `span`.

**Comment by jnurthen** [comment]
> If you move the `id` to the `em`, you won't need the bogus `span`.

They don't all have an `em`

**Comment by cookiecrook** [comment]
> > If you move the `id` to the `em`, you won't need the bogus `span`.
> 
> They don't all have an `em`

Okay keep the spans, but maybe add a comment? 

```html
<span id="step1"><!-- Don't link to this legacy numbered ID. --></span>
```

File: index.html [hunk 1]
```diff
@@ -396,17 +396,17 @@ <h4>Computation steps</h4>
...
+            <li id="comp_labelledby"><span id="step2B"><!-- Don't link to this legacy numbered ID. --></span><em>LabelledBy:</em> Otherwise, if the <code>current node</code> has an <code>aria-labelledby</code> [=attribute=] that contains at least one valid IDREF, and the <code>current node</code> is not already part of an ongoing <code>aria-labelledby</code> or <code>aria-describedby</code> traversal, process its IDREFs in the order they occur:
               <ol>
-                <li id="comp_labelledby_reset" name="step2B.i">Set the <code>accumulated text</code> to the empty string.</li>
-                <li id="comp_labelledby_foreach" name="step2B.ii">For each IDREF:
+                <li id="comp_labelledby_reset"><span id="step2B.i"><!-- Don't link to this legacy numbered ID. --></span> the <code>accumulated text</code> to the empty string.</li>
+                <li id="comp_labelledby_foreach"><span id="step2B.ii"><!-- Don't link to this legacy numbered ID. --></span> each IDREF:
```
Comment (cookiecrook) [inline index.html:402]: Looks like a few words got trampled.

```suggestion
                <li id="comp_labelledby_reset"><span id="step2B.i"><!-- Don't link to this legacy numbered ID. --></span> Set the <code>accumulated text</code> to the empty string.</li>
                <li id="comp_labelledby_foreach"><span id="step2B.ii"><!-- Don't link to this legacy numbered ID. --></span>For each IDREF:
```

File: index.html [hunk 2]
```diff
@@ -396,17 +396,17 @@ <h4>Computation steps</h4>
...
                   <ol>
-                    <li id="comp_labelledby_set_current" name="step2B.ii.a">Set the <code>current node</code> to the node referenced by the IDREF.</li>
-                    <li id="comp_labelledby_recursion" name="step2B.ii.b"><em>LabelledBy Recursion:</em> Compute the text alternative of the <code>current node</code> beginning with the overall <a href="#comp_computation">Computation</a> step. Set the <code>result</code> to that text alternative.</li>
-                    <li id="comp_labelledby_append" name="step2B.ii.c">Append a space character and the <code>result</code> to the <code>accumulated text</code>.</li>
+                    <li id="comp_labelledby_set_current"><span id="step2B.ii.a"><!-- Don't link to this legacy numbered ID. --></span> the <code>current node</code> to the node referenced by the IDREF.</li>
```
Comment (cookiecrook) [inline index.html:404]:
```suggestion
                    <li id="comp_labelledby_set_current"><span id="step2B.ii.a"><!-- Don't link to this legacy numbered ID. --></span>Set the <code>current node</code> to the node referenced by the IDREF.</li>
```

File: index.html [hunk 3]
```diff
@@ -396,17 +396,17 @@ <h4>Computation steps</h4>
...
+                    <li id="comp_labelledby_recursion"><span id="step2B.ii.b"><!-- Don't link to this legacy numbered ID. --></span><em>LabelledBy Recursion:</em> Compute the text alternative of the <code>current node</code> beginning with the overall <a href="#comp_computation">Computation</a> step. Set the <code>result</code> to that text alternative.</li>
+                    <li id="comp_labelledby_append"><span id="step2B.ii.c"><!-- Don't link to this legacy numbered ID. --></span> a space character and the <code>result</code> to the <code>accumulated text</code>.</li>
```
Comment (cookiecrook) [inline index.html:406]:
```suggestion
                    <li id="comp_labelledby_append"><span id="step2B.ii.c"><!-- Don't link to this legacy numbered ID. --></span>Append a space character and the <code>result</code> to the <code>accumulated text</code>.</li>
```

**Review by cookiecrook — APPROVED** [review]
Approve with contained diffs.

**Comment by accdc** [comment]
Is this editorial?

**Comment by jnurthen** [comment]
> Is this editorial?

yes

**Review by cookiecrook — APPROVED** [review]

**Pull request merged**